### PR TITLE
SPM: AppCenterDistribute

### PR DESCRIPTION
**Pull request by lickel**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This adds support for Swift Package Manager by the Distribute framework.

## Related PRs or issues

https://github.com/microsoft/appcenter-sdk-apple/issues/2129

## Misc

I do not know how to run the tests. I didn't see anything in the readme, nor something like a Podfile.

### CODE REVIEW

File: Package@swift-5.3.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
+// swift-tools-version:5.3
```
Comment (lickel) [inline Package@swift-5.3.swift:1]: SPM prefers the highest value tools it understands.

File: Package@swift-5.3.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                .linkedFramework("Foundation"),
+                .linkedFramework("UIKit", .when(platforms: [.iOS, .tvOS])),
+                .linkedFramework("AppKit", .when(platforms: [.macOS])),
+            ]
+        ),
+        .target(
+            name: "AppCenterCrashes",
+            dependencies: [
+                "AppCenter",
+                .product(name: "CrashReporter", package: "PLCrashReporter"),
```
Comment (lickel) [inline Package@swift-5.3.swift:72]: For whatever reason, I needed this incantation for Xcode to accept the dependency

File: Package@swift-5.3.swift [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                .linkedFramework("AppKit", .when(platforms: [.macOS])),
+            ]
+        ),
+        .target(
+            name: "AppCenterDistribute",
+            dependencies: ["AppCenter"],
+            path: "AppCenterDistribute/AppCenterDistribute",
+            exclude: ["Support"],
+            resources: [
+                .copy("AppCenterDistributeResources.bundle"),
```
Comment (AnastasiaKubova) [inline Package@swift-5.3.swift:92]: Can we use `.process` here instead `.copy`?

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## Version 4.0.1 (Under development)
 
+### App Center Distribute
+
+* **[Feature]** Support Swift Package Manager (5.3)
```
Comment (AnastasiaKubova) [inline CHANGELOG.md:7]: Could you please be mo detail about these changes?

**Review by AnastasiaKubova — COMMENTED** [review]
Hi! Thanks for creating this PR. I checked it and in general all works well.
Also, I left some comments.

**Review by MatkovIvan — APPROVED** [review]
@lickel it's a good timing, thank you!

@AnastasiaKubova let's merge it as is to the feature branch and take care of resolving feedback ourselves

**Review by AnastasiaKubova — APPROVED** [review]

**Pull request merged**